docs/user-guide/kubectl-overview.md

```diff
@@ -33,160 +33,261 @@ Documentation for other releases can be found at
 
 # kubectl overview
 
+Use this overview of the `kubectl` command line interface to help you start running commands against Kubernetes clusters. We will quickly cover `kubectl` syntax, describe the command operations, and provide common examples. For details about each command, including all the supported options, see the [kubectl](kubectl/kubectl.md) reference documentation.
+
 **Table of Contents**
 
-- [kubectl overview](#kubectl-overview)
-  - [Overview](#overview)
-  - [Common Operations](#common-operations)
-  - [Kubectl Operations](#kubectl-operations)
-  - [Resource Types](#resource-types)
+ * [Syntax](#syntax)
+ * [Operations](#operations)
+ * [Resource types](#resource-types)
+ * [Output options](#output-options)
+ * [Examples: Common operations](#examples-common-operations)
+
+TODO: Auto-generate this file to ensure it's always in sync with any `kubectl` changes, see [#14177](https://github.com/kubernetes/kubernetes/pull/14177).
+
+## Syntax
+
+Use the following syntax to run `kubectl` commands from your terminal window:
+
+```
+kubectl [command] [TYPE] [NAME] [options]
+```
+
+where `command`, `TYPE`, `NAME`, and `options` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources, for example `create`, `get`, `describe`, `delete`.
+* `TYPE`: Specifies the resource type. Both the singular and plural forms of the resource type are valid. For example, the following two commands produce the same output:  
+    ```
+    $ kubectl get pod pod1
+    $ kubectl get pods pod1
+    ```
+* `NAME`: Specifies the name of the resource. When performing an operation on multiple resources, you can specify each resource by type and name or specify one or more files:
+    * To specify resources by type and name:
+	    * To group resources if they are all the same type: `TYPE1 name1 name2 name<#>`  
+Example: `$ kubectl get pod example-pod1 example_pod2`
+		* To specify multiple resource types individually: `TYPE1/name1 TYPE1/name2 TYPE2/name3 TYPE<#>/name<#>`  
+Example: `$ kubectl get pod example-pod1 replicationcontroller example_rc1`
+	* To specify resources with one or more files: `-f file1 -f file2 -f file<#>`  
+	[Use YAML rather than JSON](config-best-practices.md) since YAML tends to be more user-friendly, especially for configuration files.  
+Example: `$ kubectl get pod -f ./pod.json`
+* `options`: Specifies optional flags. For example, you can use the `-s` or `--server` flags to specify the address and port of the Kubernetes API server.  
```

Comment:
Same question with @JanetKuo . The original page uses `[flags]` according to `kubectl-conventions.md` and `kubectl --help` output.